packages/user-extension/src/index.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+/**
+ * @packageDocumentation
+ * @module user-extension
+ */
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { IStateDB } from '@jupyterlab/statedb';
+import {
+  ICurrentUser,
+  IUserMenu,
+  RendererUserMenu,
+  User,
+  UserMenu
+} from '@jupyterlab/user';
+import { Menu, MenuBar, Widget } from '@lumino/widgets';
+
+/**
+ * A namespace for command IDs.
+ */
+export namespace CommandIDs {
+  export const settings = '@jupyterlab/user-extension:settings:open';
+}
+
+/**
+ *
+ */
+const userPlugin: JupyterFrontEndPlugin<User> = {
+  id: '@jupyterlab/user-extension:user',
+  autoStart: true,
+  requires: [IStateDB],
+  provides: ICurrentUser,
+  activate: (app: JupyterFrontEnd, state: IStateDB): User => {
+    return new User(state);
+  }
+};
+
+/**
+ *
+ */
+const userMenuPlugin: JupyterFrontEndPlugin<Menu> = {
```

Comment:
You need to define the minimal interface needed by downstream plugin - avoid providing the full menu API if possible (see later comment).

```suggestion
const userMenuPlugin: JupyterFrontEndPlugin<IUserMenu> = {
```